lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -91,6 +91,12 @@ sub _git_reset_hard ($path, $branch) {
     die "Failed to reset to 'origin/$branch': $r->{stderr}" unless $r->{status};
 }
 
+sub _git_is_workdir_clean ($path) {
+    my $r = run_cmd_with_log_return_error(['git', '-C', $path, 'diff-index', 'HEAD', '--exit-code']);
+    return $r->{status};
```

Comment:
Should we also check if there is a different kind of error?
e.g. if there is a diff, the exit code is 1, but if the command fails for some other reason, we should log this.
Also maybe we could log the actual output in case there is a diff.

Reply:
`run_cmd_with_log_return_error` is already handling this and logging any errors.